lfs.c

```diff
@@ -5233,38 +5233,67 @@ static int lfs_fs_gc_(lfs_t *lfs) {
 #endif
 
 #ifndef LFS_READONLY
+static int lfs_fs_rewrite_block_count(lfs_t *lfs, lfs_size_t block_count) {
+    lfs->block_count = block_count;
```

Comment:
I was worried about a tricky case where the superblock update could potentially relocate a superblock outside of the new filesystem bounds. But by setting `lfs->block_count` early we avoid this 👍